components/search-content/useContentSearch.tsx

```diff
@@ -0,0 +1,38 @@
+import { useLabelerAgent } from '@/shell/ConfigurationContext'
+import { useWorkspaceAddItemsMutation } from '@/workspace/hooks'
+import { useInfiniteQuery } from '@tanstack/react-query'
+
+export const useContentSearch = ({
+  term,
+  section,
+}: {
+  term: string
+  section: string
+}) => {
+  const labelerAgent = useLabelerAgent()
+
+  const { mutate: addToWorkspace } = useWorkspaceAddItemsMutation()
+
+  const searchInfo = useInfiniteQuery({
+    queryKey: ['searchContent', { term, section }],
+    enabled: !!term,
+    queryFn: async ({ pageParam }) => {
+      const { data } = await labelerAgent.app.bsky.feed.searchPosts({
+        q: term,
+        limit: 30,
+        sort: section,
+        cursor: pageParam,
+      })
+      return data
+    },
+    getNextPageParam: (lastPage) => lastPage.cursor,
+  })
```

Comment:
I fear that this query could have two effects:
1) a lot of entries end up being cached, significantly increasing the memory usage
2) there isn't any mechanism to clear that cache so the user could end-up seeing stale results when trying to search for newer posts.

Is there a way to invalidate all `searchContent` cache entries, for example when the component gets destroyed ?

Alternatively, maybe a mutation would be more appropriate here ?

Reply:
good point but I think this is actually what we want. when investigating search results, I don't think a 5m delay in seeing realtime content is a big issue for mods and in fact, we wouldn't want the results to change while they are scanning a specific page for a keyword so overall, the caching seems to be a good idea.
keep in mind that at any given moment, a mod will probably scan for only a few keywords so keyword specific caches can be kept around without overwhelming memory usage IMO.